tee-worker/core-primitives/types/src/lib.rs

```
@@ -91,11 +90,26 @@ impl Encode for OpaqueCall {
 	}
 }
 
-// Note in the pallet teerex this is a struct. But for the codec this does not matter.
-#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
-pub struct Request {
-	pub shard: ShardIdentifier,
-	pub cyphertext: Vec<u8>,
+// Litentry: re-declared due to orphan rule (that's why macro is used)
+decl_rsa_request!(Debug);
+
+impl DecryptableRequest for RsaRequest {
+	type Error = ();
+
+	fn shard(&self) -> ShardIdentifier {
+		self.shard
+	}
+
+	fn payload(&self) -> &[u8] {
+		self.payload.as_slice()
+	}
+
+	fn decrypt<T: Debug>(
+		&mut self,
+		enclave_shielding_key: Box<dyn ShieldingCryptoDecrypt<Error = T>>,
+	) -> core::result::Result<Vec<u8>, ()> {
+		enclave_shielding_key.decrypt(self.payload.as_slice()).map_err(|_| ())
```

Not sure I follow, what error is being swallowed here?

---

The `T` here : ) - we don't care about this error, as in the end they will be converted to the top_pool decryption error anyway